clippy_lints/src/needless_borrowed_ref.rs

```diff
@@ -43,12 +43,11 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for NeedlessBorrowedRef {
         }
 
         if_let_chain! {[
-            // Pat is a pattern whose node
-            // is a binding which "involves" a immutable reference...
+            // Pat is a pattern whose node is a binding which "involves" an immutable reference...
             let PatKind::Binding(BindingMode::BindByRef(MutImmutable), ..) = pat.node,
             // Pattern's type is a reference. Get the type and mutability of referenced value (tam: TypeAndMut).
             let ty::TyRef(_, ref tam) = cx.tables.pat_ty(pat).sty,
-            // This is an immutable reference.
+            // Only lint immutable refs, because `&mut ref T` may be useful.
```

Comment:
Please add a test that we do not lint in this case. (can be in the same file as the other test)